tests/test_builders/test_build_linkcheck.py

```diff
@@ -1439,3 +1439,87 @@ def test_linkcheck_exclude_documents(app: SphinxTestApp) -> None:
         'uri': 'https://www.sphinx-doc.org/this-is-another-broken-link',
         'info': 'br0ken_link matched br[0-9]ken_link from linkcheck_exclude_documents',
     } in content
+
+
+class CaseSensitiveHandler(BaseHTTPRequestHandler):
+    """Simple test server for case sensitivity tests."""
+
+    protocol_version = 'HTTP/1.1'
+
+    def do_HEAD(self):
+        if self.path == '/path':
+            # Redirect lowercase /path to uppercase /Path
+            self.send_response(301, 'Moved Permanently')
+            self.send_header('Location', '/Path')
+            self.send_header('Content-Length', '0')
+            self.end_headers()
+        elif self.path == '/Path':
+            self.send_response(200, 'OK')
+            self.send_header('Content-Length', '0')
+            self.end_headers()
+        else:
+            self.send_response(404, 'Not Found')
+            self.send_header('Content-Length', '0')
+            self.end_headers()
+
+    def do_GET(self):
+        if self.path == '/path':
+            # Redirect lowercase /path to uppercase /Path
+            self.send_response(301, 'Moved Permanently')
+            self.send_header('Location', f'http://{self.headers["Host"]}/Path')
+            self.send_header('Content-Length', '0')
+            self.end_headers()
+        elif self.path == '/Path':
+            content = b'ok\n\n'
+            self.send_response(200, 'OK')
+            self.send_header('Content-Length', str(len(content)))
+            self.end_headers()
+            self.wfile.write(content)
+        else:
+            self.send_response(404, 'Not Found')
+            self.send_header('Content-Length', '0')
+            self.end_headers()
+
+
+@pytest.mark.sphinx(
+    'linkcheck',
+    testroot='linkcheck-case-check',
+    freshenv=True,
+    confoverrides={'linkcheck_case_sensitive': True},
+)
+def test_linkcheck_case_sensitive(app: SphinxTestApp) -> None:
+    """Test that case-sensitive checking is the default behavior."""
+    with serve_application(app, CaseSensitiveHandler) as address:
+        app.build()
+
+    content = (app.outdir / 'output.json').read_text(encoding='utf8')
+    rows = [json.loads(x) for x in content.splitlines()]
+    rowsby = {row['uri']: row for row in rows}
+
+    # With case-sensitive checking, a URL that redirects to different case
+    # should be marked as redirected
+    lowercase_uri = f'http://{address}/path'
+    assert lowercase_uri in rowsby, f'Expected {lowercase_uri} to be checked'
+    assert rowsby[lowercase_uri]['status'] == 'redirected'
```

Comment:
Although the containment check is good defensive coding, we don't perform similar checks in the other tests, so for consistency I would recommend:

```suggestion
    assert rowsby[f'http://{address}/path']['status'] == 'redirected'
```

(I briefly considered suggesting renaming the variable to something like `origin_url`, `outbound_url` or `documented_url` -- but on balance I think we should simply omit it)

Reply:
I've simplified both tests to directly assert on `rowsby[f'http://{address}/path']['status']` without the intermediate variable or defensive containment check, matching the style used in other tests in the file